### PR TITLE
Ingester binaries & Docker

### PR DESCRIPTION
**Pull request by ledor473**
## Which problem is this PR solving?
Resolves https://github.com/jaegertracing/jaeger/issues/1031 

## Short description of the changes
This supersede https://github.com/jaegertracing/jaeger/pull/1033

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -198,6 +198,14 @@ build-query:
 build-collector:
 	CGO_ENABLED=0 installsuffix=cgo go build -o ./cmd/collector/collector-$(GOOS) $(BUILD_INFO) ./cmd/collector/main.go
 
+.PHONY: build-ingester
+build-ingester:
+	CGO_ENABLED=0 installsuffix=cgo go build -o ./cmd/ingester/ingester-$(GOOS) $(BUILD_INFO) ./cmd/ingester/main.go
+
+.PHONY: build-ingester-linux
```
Comment (ledor473) [inline Makefile]: This was added by @zoidbergwill but there's no `build-collector-linux` nor a `build-agent-linux`, so I see 2 options:
1. I remove this and we recommend using `GOOS=linux make build-ingester`
2. I add one for every binary

Comment (zoidbergwill) [reply Makefile]: Ahhh, I see now, only the all in one has a `-linux` job. Recommending `GOOS=linux make build-ingester` would definitely be good enough. I thought I'd found some other ones that did the same, my bad.

Comment (ledor473) [reply Makefile]: Hum I missed that one as well.

**Comment by ledor473** [comment]
DCO complains about the commit from @zoidbergwill so I'm not sure how to handle that.
>Expected "William Stewart william.stewart@booking.com", but got "William Stewart zoidbergwill@gmail.com"

**Comment by zoidbergwill** [comment]
Whoops, those are both me. What's the easiest way to fix it? Could I rebase that commit and fix my sign-off / author e-mail?

**Comment by zoidbergwill** [comment]
Fixed it in my fork: https://github.com/zoidbergwill/jaeger/commit/8581a8b4861a3a4cc555aa440e8f62cd762cdfd9.patch

**Comment by ledor473** [comment]
Thanks @zoidbergwill 👍 I've rebase my change and DCO is ✅ now.

**Comment by yurishkuro** [comment]
>  zoidbergwill

+1

**Comment by yurishkuro** [comment]
futurama refs are 1st class citizen in this repo

File: Makefile (hunk 1 above)
Comment (ledor473) [reply Makefile]: It's used by `./scripts/travis/build-all-in-one-image.sh`, but I can add that cleanup to the PR

File: cmd/ingester/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM alpine:latest as certs
+RUN apk add --update --no-cache ca-certificates
+
+FROM scratch
+
+COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
```
Comment (pavolloffay) [inline cmd/ingester/Dockerfile]: are certificates needed?

**Comment by pavolloffay** [comment]
> Thanks @zoidbergwill +1 I've rebase my change and DCO is white_check_mark now.

@ledor473 maybe forgot to push?

**Comment by ledor473** [comment]
Seems like I forgot to sign my 2nd commit indeed. It's fixed now

File: cmd/ingester/Dockerfile (hunk 2 above)
Comment (yurishkuro) [reply cmd/ingester/Dockerfile]: at this point prob not, but I assume it's feasible that people might use certs to auth to Kafka.

**Comment by yurishkuro** [comment]
should this be rebased or merge master in? 'cause there's a fix for Kafka tests in master.

**Comment by ledor473** [comment]
Rebased, squashed and confirmed that I've signed my commit 👍

**Comment by yurishkuro** [comment]
Awesome!

**Pull request merged**